src/platform/qt/GBAApp.cpp

```diff
@@ -163,9 +163,9 @@ QString GBAApp::getOpenFileName(QWidget* owner, const QString& title, const QStr
 	QString filename = QFileDialog::getOpenFileName(owner, title, m_configController.getOption("lastDirectory"), filter);
 	continueAll(paused);
 	if (!filename.isEmpty()) {
-		m_configController.setOption("lastDirectory", QFileInfo(filename).dir().path());
+		m_configController.setOption("lastDirectory", QFileInfo(filename).dir().canonicalPath());
 	}
-	return filename;
+	return QDir::toNativeSeparators(QFileInfo(filename).canonicalFilePath());
```

Comment:
Why are you returning a native separator filename to the program if this is used for internal purposes, not user feedback?